koku/masu/external/downloader/azure/azure_service.py

```diff
@@ -0,0 +1,96 @@
+#
+# Copyright 2019 Red Hat, Inc.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+#
+"""Azure Service helpers."""
+from tempfile import NamedTemporaryFile
+
+from providers.azure.client import AzureClientFactory
+
+
+class AzureCostReportNotFound(Exception):
+    """Raised when Azure cost report is not found."""
+
+    pass
+
+
+class AzureService:
+    """A class to handle interactions with the Azure services."""
+
+    def __init__(self, subscription_id, tenant_id, client_id, client_secret,
+                 resource_group_name, storage_account_name, cloud='public'):
+        """Establish connection information."""
+        self._factory = AzureClientFactory(subscription_id, tenant_id, client_id, client_secret, cloud)
+        self._cloud_storage_account = self._factory.cloud_storage_account(resource_group_name, storage_account_name)
+
+        if not self._factory.credentials:
+            raise ValueError('Azure Service credentials are not configured.')
+
+    def get_cost_export_for_key(self, key, container_name):
+        """Get the latest cost export file from given storage account container."""
+        report = None
+        cloud_storage_account = self._cloud_storage_account
+        blockblob_service = cloud_storage_account.create_block_blob_service()
+        blob_list = blockblob_service.list_blobs(container_name)
```

Comment:
Should the calls to `create_block_blob_service()` or `list_blobs` be wrapped in a try/except block?

They're remote API calls that could raise an `AzureException`. Is that going to be caught anywhere?

Reply:
yeah thats a good point.  I'll add that